### ISSUE TITLE
When link has no meta it should render link as shortened version ("rel": "href")

### ISSUE THREAD

**Issue by marceloverdijk** [jasminb/jsonapi-converter#182]
Currently links are rendered as:

```
{
  "links": {
    "self": {
      "href": "abc",
      "meta": {
        "foo": "bar"
      }
    }
  }
}
```

and without meta as:

```
{
  "links": {
    "self": {
      "href": "abc",
      "meta": null
    }
  }
}
```

or in case https://github.com/jasminb/jsonapi-converter/pull/181 is accepted as:

```
{
  "links": {
    "self": {
      "href": "abc"
    }
  }
}
```

Still not perfect. In case no meta is available I think rendering it to the below sample would be preferable:

```
{
  "links": {
    "self": "abc"
  }
}
```

This is also done like this in the jsonapi specs.

We could solve this by writing a custom `JsonSerializer` for the `Link` object.

Would you be interested in a PR for that?

**Comment by jasminb** [comment]
Quick thought, maybe this could be solved with meta-aware link getters in combination with disabling null value serialisation?

**Comment by marceloverdijk** [comment]
In general I *do* like null properties to be serialized, especially for the `”attributes”` element of the resources.

But for the core jsonapi elements I don’t think it’s usefull, like the `”meta”` element.
And that’s also why I would prefer the shorthand version of the link if possible.

PS: I know I’m not consistent regarding serializing null values here 😇 (but at least it’s how I would prefer the output to look like)

**Comment by marceloverdijk** [comment]
Closed as fixed by PR #188
